environments/environment.py

```diff
@@ -77,6 +77,38 @@ def get_default_radius_bounds(self):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def _sample_init_conditions(self, radius_bound):
+        """Samples random initial conditions for the environment
+
+        Args:
+            radius_bound (float, float): Radius lower and upper bound of the phase state sampling.
+                Optionally, it can be a string 'auto'. In that case, the value returned by
+                get_default_radius_bounds() will be returned.
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def _get_default_ball_color(self):
+        """Returns:
+            color_ball (tuple): (R, G, B) default color ball for rendering.
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
+
+    def _get_default_background_color(self):
```

Comment:
I think its better if backround_color is a protected class member, instead of a function that returns a hard-coded value

Reply:
you mean self._background_color right?

Reply:
yes